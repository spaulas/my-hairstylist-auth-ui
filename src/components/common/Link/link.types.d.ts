export type LinkProps = {
  url: string;
  title: string;
  className?: string;
};
