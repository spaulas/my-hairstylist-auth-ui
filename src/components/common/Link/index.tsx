import "./link.styles.scss";
import { LinkProps } from "./link.types";

const Link = ({
  url,
  title,
  className = "",
}: LinkProps): React.ReactElement => {
  return (
    <a className={`link ${className}`} href={url}>
      {title}
    </a>
  );
};

export default Link;
