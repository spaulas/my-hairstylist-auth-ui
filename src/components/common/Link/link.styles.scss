@import "@styles/imports";

.link {
  @include link;
  cursor: pointer;
  color: $primary--blue;
  text-align: center;
  transition: all 0.2s ease;

  &:hover {
    color: $primary--blue__medium;
  }
}
