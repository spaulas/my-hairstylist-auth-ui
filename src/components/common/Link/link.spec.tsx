import "@testing-library/jest-dom/extend-expect";
import Link from ".";
import { render, screen } from "@testing-library/react";

describe("Link", () => {
  const props = {
    title: "Sign up",
    url: "/auth/sign-up",
    className: "className",
  };

  it("renders correctly", () => {
    const { container } = render(<Link {...props} />);
    expect(container).toMatchSnapshot();
  });

  describe("className", () => {
    it("should have props.className when defined", () => {
      render(<Link {...props} />);
      const link = screen.getByRole("link");

      expect(link).toHaveClass("link");
      expect(link).toHaveClass(props.className);
    });

    it("should be empty when defined", () => {
      const noClassName = { ...props, className: undefined };
      render(<Link {...noClassName} />);
      const link = screen.getByRole("link");

      expect(link).toHaveClass("link");
    });
  });

  it("should display title", () => {
    render(<Link {...props} />);
    const title = screen.getByText(props.title);
    expect(title).toBeVisible();
  });

  it("should have href", () => {
    render(<Link {...props} />);
    const link = screen.getByRole("link");

    expect(link).toHaveAttribute("href", props.url);
  });
});
