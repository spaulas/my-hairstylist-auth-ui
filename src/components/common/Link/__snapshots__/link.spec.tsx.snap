// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div>
  <a
    class="link className"
    href="/auth/sign-up"
  >
    Sign up
  </a>
</div>
`;
